### PR TITLE
Add buffer to callback to node api

### PR DESCRIPTION
**Pull request by gregorycuellar**
This is a small modification to pass the sound buffer to the event callback method.

### CODE REVIEW

**Comment by evancohen** [comment]
Good idea! I'm wondering if this will have a negative performance impact on low powered devices like the Pi and the CHIP.

I did some experimenting with something similar to this in [`sonus/audio-buffer`](https://github.com/evancohen/sonus/tree/audio-buffer) with a ring buffer to remove the gap between a spoken hotword and when audio starts streaming for detection. The results were awesome, but it also lead to a lag in detection by about 1/3 to 1/5 a second.

Maybe it was just a bad implementation on my part, but in my experience the additional buffer has a pretty substantial impact on performance. I'll test out this change and do some controlled tests when I get the chance to see what the real impact is.

**Comment by chenguoguo** [comment]
@gregorycuellar thanks a lot for the PR! Did you see the performance issue that @evancohen mentioned?

@evancohen please let me know if you see the performance issue after testing this.

The PR itself looks good to me, I'll wait for a few days for the performance test :-)

File: examples/Node/file.js [hunk 1]
```diff
@@ -21,8 +21,8 @@ detector.on('silence', function () {
   console.log('silence');
 });
 
-detector.on('sound', function () {
-  console.log('sound');
+detector.on('sound', function (buffer) {
+  console.log('sound', buffer);
```
Comment (chenguoguo) [inline examples/Node/file.js]: Perhaps we don't have to send the buffer data to console? I'm just afraid that people may get scared by the binary data from console :-) Perhaps, adding a comment of how people can possibly use this buffer data will be enough?

File: examples/Node/microphone.js [hunk 2]
```diff
@@ -20,8 +20,8 @@ detector.on('silence', function () {
   console.log('silence');
 });
 
-detector.on('sound', function () {
-  console.log('sound');
+detector.on('sound', function (buffer) {
+  console.log('sound', buffer);
```
Comment (chenguoguo) [inline examples/Node/microphone.js]: Same as above, let's not send the binary data to the console. Let's use a comment here.

**Comment by gregorycuellar** [comment]
I made some quick test for performance, and so far, I didn't saw something.
As it's just passing parameter by reference, I think, it's not going to have an impact on performance.
But maybe it depends on the callback method implementation. Tell me, if you have some hint where to search ;-)

it's important matter, so I will do more test to check, specialy on rpi.

File: examples/Node/file.js (hunk 1 above)
Comment (gregorycuellar) [inline examples/Node/file.js]: You are right, as it's an example, I left the console.log for demo... If I remember well, it's not showing binary, but object reference ( Maybe it depends on node versions )
Anyway, it's not good to leave console.log ;-)

**Comment by evancohen** [comment]
I haven't had the chance to look at this yet, but I'm guessing @gregorycuellar is right - since it's only being passed by reference it shouldn't have a performance impact.

File: examples/Node/file.js (hunk 1 above)
Comment (chenguoguo) [inline examples/Node/file.js]: @gregorycuellar could you address the two comments I left here? I'll be ready to merge after that.

**Comment by chenguoguo** [comment]
Thanks guys! Merging it!

**Pull request merged**